airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -224,6 +224,7 @@ def _group_streams(
                     stream_state = self._connector_state_manager.get_stream_state(
                         stream_name=declarative_stream.name, namespace=declarative_stream.namespace
                     )
+                    stream_state = self._migrate_state(declarative_stream, stream_state)
```

Comment:
nit: can we move the `self._migrate_state` call inside `_get_retriever as we only do that for the retriever part?

Reply:
for me it's more logic for `_connector_state_manager `, but if we don't want to change its method declaration I prefer to override `stream_state ` variable as we will be sure that next using of `stream_state` contains migrated and actual state.